Freshness for the authentication cmdlets article (#694)
MicrosoftDocs/microsoftgraph-docs-powershell: microsoftgraph/docs-conceptual/authentication-commands.md

#### Use delegated access with a custom application for Microsoft Graph PowerShell

> Follow the steps below to create a custom application that you can use to connect to Microsoft Graph PowerShell. Use this approach if you need to isolate and limit the consent permissions granted for Microsoft Graph PowerShell usage.

Reviewer: ```suggestion
Complete the following steps to create a custom application that you can use to connect to Microsoft Graph PowerShell. Use this approach if you need to isolate and limit the consent permissions granted for Microsoft Graph PowerShell usage.
```

#### Use client credential with a certificate

> To use app-only access, you can load the certificate from either *Cert:\CurrentUser\My\\* or *Cert:\LocalMachine\My\\* when you specify the `-CertificateThumbprint` or `-CertificateName` parameters with `Connect-MgGraph`. Make sure that the certificate you're using is present in either certificate store before calling `Connect-MgGraph`. For more info, see [Use app-only authentication with the Microsoft Graph PowerShell SDK](app-only.md).

Reviewer: ```suggestion
To use app-only access, you can load the certificate from either *Cert:\CurrentUser\My\\* or *Cert:\LocalMachine\My\\* when you specify the `-CertificateThumbprint` or `-CertificateName` parameters with `Connect-MgGraph`. Ensure the certificate you're using is present in either certificate store before calling `Connect-MgGraph`. For more info, see [Use app-only authentication with the Microsoft Graph PowerShell SDK](app-only.md).
```

#### Use client secret credentials

> This type of grant will help when you need interactions in the background without a user to sign in. You can use the **-ClientSecretCredential** parameter with **Connect-MgGraph** to provide client secret credentials. See [Get-Credential](/powershell/module/microsoft.powershell.security/get-credential) on how to get or create credentials.

Reviewer: ```suggestion
This type of grant helps when you need interactions in the background without a user to sign in. You can use the **-ClientSecretCredential** parameter with **Connect-MgGraph** to provide client secret credentials. See [Get-Credential](/powershell/module/microsoft.powershell.security/get-credential) on how to get or create credentials.
```

```

>[!NOTE]
>It's recommended to use PowerShell 7 and above when using client secret credentials, as it provides improved security features and better compatibility with modern authentication methods.

#### Use managed identity

A common challenge when writing automation scripts is managing secrets, credentials, certificates, and keys. These are used to secure communication between services. Eliminate the need to manage credentials by allowing the module to obtain access tokens for Microsoft Entra resources that are protected by Microsoft Entra ID. The identity is managed by the Microsoft Entra platform and does not require you to provision or rotate any secrets.

> A common challenge when writing automation scripts is the management of secrets, credentials, certificates, and keys used to secure communication between services. Eliminate the need to manage credentials by allowing the module to obtain access tokens for Microsoft Entra resources that are protected by Microsoft Entra ID. The identity is managed by the Microsoft Entra platform and does not require you to provision or rotate any secrets.

Reviewer: ```suggestion
A common challenge when writing automation scripts is the management of secrets, credentials, certificates, and keys used to secure communication between services. Eliminate the need to manage credentials by allowing the module to obtain access tokens for Microsoft Entra resources that are protected by Microsoft Entra ID. The identity is managed by the Microsoft Entra platform and doesn't require you to provision or rotate any secrets.
```

- User-assigned managed identity: Uses a user created managed identity as a standalone Microsoft Entra resource.

    ```powershell
    Connect-MgGraph -Identity -ClientId "<USER_ASSIGNED_MANAGED_IDENTITY_CLIENT_ID>"
    ```

### Connect to an environment or cloud

> By default, `Connect-MgGraph` targets the global public cloud. For more details on targeting other clouds, see the [Use Get-MgEnvironment](#use-get-mgenvironment) section below.

Reviewer: ```suggestion
By default, `Connect-MgGraph` targets the global public cloud. For more details on targeting other clouds, see the [Use Get-MgEnvironment](#use-get-mgenvironment) section.
```

Microsoft Graph PowerShell supports the following passwordless authentication methods:

- Windows Hello for Business
- Fast Identity Online v2.0 (FIDO2)
- Microsoft Authenticator app
- Certificate-based authentication (CBA)

>[!NOTE]
> The use of FIDO2 security keys for authentication is supported only in PowerShell 7 and above.

For more information, see [Passwordless authentication options for Microsoft Entra ID](/azure/active-directory/authentication/concept-authentication-passwordless) and [Microsoft Entra ID certificate-based authentication](/azure/active-directory/authentication/concept-certificate-based-authentication).

## Use Disconnect-MgGraph

> Once you're signed in, you'll remain signed in until you invoke `Disconnect-MgGraph`. Microsoft Graph PowerShell automatically refreshes the access token for you, and sign-in persists across PowerShell sessions because Microsoft Graph PowerShell securely caches the token when using the default `CurrentUser` context scope. If you use the `-ContextScope Process` parameter with `Connect-MgGraph`, sign-in will only persist for the current PowerShell session.

Reviewer: ```suggestion
Once you're signed in, you remain signed in until you invoke `Disconnect-MgGraph`. Microsoft Graph PowerShell automatically refreshes the access token for you, and sign-in persists across PowerShell sessions because Microsoft Graph PowerShell securely caches the token when using the default `CurrentUser` context scope. If you use the `-ContextScope Process` parameter with `Connect-MgGraph`, sign-in only persists for the current PowerShell session.
```

USGovDoD https://login.microsoftonline.us  https://dod-graph.microsoft.us          Built-in
```

To explicitly target other clouds, for example, US Government and Azure China, use the `-Environment` parameter, which accepts values shown in the previous output (such as `Global`, `China`, `USGov`, and `USGovDoD`).

```powershell
Connect-MgGraph -Environment USGov
```

>[!NOTE]
>[!IMPORTANT]

> >Globally registered apps do not replicate to Azure China. You must [register your own applications](/entra/identity-platform/quickstart-register-app) and use them when connecting to Microsoft Graph.

Reviewer: ```suggestion
>Globally registered apps don't replicate to Azure China. You must [register your own applications](/entra/identity-platform/quickstart-register-app) and use them when connecting to Microsoft Graph.
```

ClientTimeout         : 00:05:00
```

To retrieve all the scopes that you've consented to, use `Select-Object` with the `-ExpandProperty` parameter to expand the `Scopes` property:

> Each scope will be output on a new line when using `-ExpandProperty`.

Reviewer: ```suggestion
Each scope is output on a new line when using `-ExpandProperty`.
```
